tests/attribute_annotations.pyi

```diff
@@ -1,16 +1,31 @@
-field1: int
-field2: int = ...
+from typing import TypeAlias
+
+field0: int
+field_foo: int = ...  # Y032 Default value unnecessary. Use "field_foo: int" instead of "field_foo: int = ..."
+field_bar: int = 0  # Y032 Default value unnecessary. Use "field_bar: int" instead of "field_bar: int = 0"
```

Comment:
I don't like how this accounts for class members that just happen to use the same name. I find it confusing, and it seems like it will make refactoring stubs fragile and surprising.

Has anyone looked into disabling the flake8 undefined variable warning yet? Alternatively, would it be less work to not count instance attributes, argument names etc as using the global variable?

Reply:
Yeah, this bothers me too. I initially tried to work around it (see my first commit on this PR), but it was more complicated than I expected. I have another idea for a fix, which I'll try tomorrow.